### PR TITLE
feat: automated releases and auto labelling

### PR DESCRIPTION
**Pull request by jmeridth**
Part of https://github.com/github/github-ospo/issues/78

# Pull Request

## Proposed Changes

- [x] lint pr title adhering to conventional commits (helps with auto labelling)
  - [x] Update CONTRIBUTING.md
    - [x] automated releases
    - [x] how to file a bug report
    - [x] how to file an enhancement
  - [x] Update pull_request_template with new standard
- [x] setup autolabelling through release-drafter github action config
- [x] create auto-labeler workflow
- [x] change release-drafter workflow to release
  - [x] create release, including new tag
  - [x] create discussion announcement based on release
  - [x] build container images, tag them and release them

Manual Actions:

- [x] create `revert` and `maintenance` labels
- [x] put repository and category IDs into secrets for the discussion create action

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, or `breaking`

### CODE REVIEW

File: .github/pull_request_template.md
Comment (zkoppert) [inline .github/pull_request_template.md:1]: Need to change line 24 as well here to match pr labels like maintenance?

**Review by zkoppert — APPROVED** [review]
I think these releases may silently fail on major version changes since we aren't updating action.yaml files. That sound right to you? We can make that a future pr (set of prs) if desired.

**Comment by jmeridth** [comment]
> I think these releases may silently fail on major version changes since we aren't updating action.yaml files. That sound right to you? We can make that a future pr (set of prs) if desired.

yes, currently, we, as the PR reviewers, need to confirm the action.yml was updated if we're going to flag a PR with the `breaking` label.  We can automate that though.  Definitely a new PR.

Do you think this work is a major version bump?  I haven't been doing that. 😬

**Comment by zkoppert** [comment]
No major bump needed for this one, just looking to the future.

**Pull request merged**